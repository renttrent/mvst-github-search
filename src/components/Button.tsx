import { MouseEventHandler } from "react";

interface ButtonProps {
  label: string;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  className?: string;
  style?: Object;
}

/**
 * Button
 *
 * @param onClick Search click handler, set by parent component
 * @returns JSX elemnt
 */
export const Button: React.FC<ButtonProps> = ({
  label,
  className = undefined,
  ...props
}) => {
  return (
    <button
      className={
        className ||
        "bg-indigo-500 hover:bg-indigo-700 text-white p-2 rounded-md w-full font-semibold"
      }
      {...props}
    >
      {label}
    </button>
  );
};
