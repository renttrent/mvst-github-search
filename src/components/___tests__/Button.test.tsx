import { render, screen } from "@testing-library/react"
import { Button } from "../Button"

// it("should render Button normally", () => {
//   const component = renderer.create(<Button label="Search" />)
//   let tree = component.toJSON()
//   expect(tree).toMatchSnapshot()

//   renderer.act(() => {
//     tree.props.onMouseEnter()
//   })

//   tree = component.toJSON()
//   expect(tree).toMatchSnapshot()

//   renderer.act(() => {
//     tree.props.onMouseLeave()
//   })
//   tree = component.toJSON()

//   expect(tree).toMatchSnapshot()
// })