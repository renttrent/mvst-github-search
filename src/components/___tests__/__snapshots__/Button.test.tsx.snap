// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Button normally 1`] = `
<button
  className="bg-indigo-500 hover:bg-indigo-700 text-white p-2 rounded-md w-full font-semibold"
>
  Search
</button>
`;
